src/storage/db_interface_common.py

```diff
@@ -310,3 +311,50 @@ def release_unpacking_lock(self, uid):
 
     def drop_unpacking_locks(self):
         self.main.drop_collection('locks')
+
+    def _collect_analysis_tags_from_children(self, uid):
+        children = self._fetch_children_with_tags(uid)
+        unique_tags = {}
+        for child in children:
+            self._collect_analysis_tags(child, unique_tags)
+        return unique_tags
+
+    def _collect_analysis_tags(self, file_object, analysis_tags):
+        for name, analysis in [(n, a) for n, a in file_object.processed_analysis.items() if 'tags' in a]:
+            if 'file_system_flag' in analysis and analysis['file_system_flag']:
+                analysis = self.retrieve_analysis(file_object.processed_analysis, analysis_filter=[name, ])[name]
```

Comment:
why `[name, ]` and not just `[name]`?

Reply:
Nagging insecurity. Once I defined sth. similar to `['string']` and when I accessed it's first element I got `'s'`.
You want me to change it?